OPC UA chapter for troubleshooting enhanced, by adding JMX (#497)
SoftwareAG/c8y-docs: content/protocol-integration/opcua-bundle/troubleshooting.md

```

For additional information about log levels, refer to the [Logback architecture documentation](http://logback.qos.ch/manual/architecture.html#effectiveLevel).

### Java Management Extensions (JMX) 

For additional monitoring, the Gateway component provides MBeans. Those MBeans get exposed if following configuration is set in the application.yaml

```
spring:
     jmx:
         enabled: true
```

> Via jconsole the MBeans can be selected and following attributes can be accessed:

Reviewer: ```suggestion
Via jconsole the MBeans can be selected and the following attributes can be accessed:
```

> ![jconsole MBeans](/images/device-protocols/opcua/opcua-jmx-mbeans.png)

Reviewer: the image files use ".PNG" file extension. this should be lower case. the image links should also then use lower case (it's correct here, but there are two more occurrences where it's upper case).

Author: i changed the other to lower case

> Particular for custom actions it is usefull to get some statistics. This attributes can be retrieved from CustomActionMBean:

Reviewer: ```suggestion
It can be is useful to get some statistics for custom actions in particular. These attributes can be retrieved from the CustomActionMBean:
```

> 1. Table of all called URLs seperated by http return code and retry count.

Reviewer: ```suggestion
1. Table of all called URLs seperated by HTTP return code and retry count.
```

> ![jconsole MBeans CustomActionMBean CounterTable](/images/device-protocols/opcua/opcua-jmx-customActionMBean-CounterTable.PNG)

Reviewer: since we're in an enumeration here, i'm not sure about the indentation. i can't check the PR out to have a look at it locally, also see my question about the fork.

> The key entry of the table contains of 

Reviewer: ```suggestion
The key entry of the table consists of:
```

```
{URL}_{HTTP Response Code}_{Retry Count}
```

> 

Reviewer: ```suggestion
```

```

For additional information about log levels, refer to the [Logback architecture documentation](http://logback.qos.ch/manual/architecture.html#effectiveLevel).

### Java Management Extensions (JMX) 

For additional monitoring, the Gateway component provides MBeans. Those MBeans get exposed if following configuration is set in the application.yaml

```
spring:
     jmx:
         enabled: true
```

Via jconsole the MBeans can be selected and following attributes can be accessed:

![jconsole MBeans](/images/device-protocols/opcua/opcua-jmx-mbeans.png)

Particular for custom actions it is usefull to get some statistics. This attributes can be retrieved from CustomActionMBean:

1. Table of all called URLs seperated by http return code and retry count.

![jconsole MBeans CustomActionMBean CounterTable](/images/device-protocols/opcua/opcua-jmx-customActionMBean-CounterTable.PNG)

The key entry of the table contains of 

```
{URL}_{HTTP Response Code}_{Retry Count}
```

2. If retry is enabled, the queue size of the retry queue can be monitored.

![jconsole MBeans CustomActionMBean RetryQueueSize](/images/device-protocols/opcua/opcua-jmx-customActionMBean-RetryQueueSize.PNG)

> ### Prometheus

Reviewer 2: We do not document what we don't have :) Should be removed.

Author: Do you mean the complete Prometheus part?
